### DISCUSSION TITLE
Adding Auto-Completion support for .toml metadata files

### DISCUSSION THREAD

**Discussion by nadeeshaan** [ballerina-platform/ballerina-lang#29304] [Ideas]
Currently we do not have auto-completion support for the .toml metadata files in ballerina except the Cloud.toml. What if we have a generic completion resolver for the toml completions. Since we have the support for Cloud.toml already we can have a generic layer for toml.

Currently, the language server's completion engine is extendable and we can register providers without any hassle. 
@xlight05 @hasithaa @IMS94 @rasika

**Comment by xlight05** [comment]
In Cloud.toml auto completion implementation, completion items generated using a json schema defined for the cloud spec . We should be able to achieve this by making this a util.
Also currently this code resides in c2c repo. Should we add this part to lang?

**Reply by nadeeshaan** [reply]
We might need to think of an approach which is customisable in the sense of json schema as well as any additional capabilities. In the future we will have to incorporate snippets and other special capabilities. 
WDYT?

**Reply by rasika** [reply]
@xlight05 that's great! I think for the first cut; we can allow completions for the other `.toml` files.  If the completion items for the Cloud.toml are generated from a json schema we can make it generic with minimum effort for other files as well.

**Reply by xlight05** [reply]
@rasika yeah, afaik, ballerina.toml and dependancies.toml already has a json schema defined for validation purposes as well.

@nadeeshaan yes we need to make sure if all the requirements can be covered from json schema.
